test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetCurrentContendedMonitor/contmon001.java

```diff
@@ -55,8 +54,17 @@ public static void main(String[] args) {
         System.exit(run(args, System.out) + 95/*STATUS_TEMP*/);
     }
 
+    public static void doSleep() {
+        try {
+            Thread.sleep(10);
+        } catch (Exception e) {
+            throw new Error("Unexpected " + e);
+        }
+    }
+
     public static int run(String argv[], PrintStream ref) {
         out = ref;
+        doSleep(); // If it would do any class loading, do it now.
```

Comment:
I think now the spawned thread should not try to resolve any new methods after setting the boolean so we shouldn't have the same ObjectLocker issue.